packages/playground/website/src/components/playground-viewport/index.tsx

```diff
@@ -21,52 +28,110 @@ interface PlaygroundViewportProps {
 	className?: string;
 }
 
-export const PlaygroundViewport = React.forwardRef<
-	HTMLIFrameElement,
-	PlaygroundViewportProps
->(
-	(
-		{ displayMode = 'browser', toolbarButtons, hideToolbar, className },
-		ref
-	) => {
-		const { playground, currentUrl: url } = usePlaygroundContext();
+export const PlaygroundViewport = ({
+	displayMode = 'browser-full-screen',
+	toolbarButtons,
+	hideToolbar,
+	className,
+}: PlaygroundViewportProps) => {
+	const clientInfo = useAppSelector(getActiveClient);
+	const playground = clientInfo?.client;
+	const url = clientInfo?.url;
 
-		if (displayMode === 'seamless') {
-			// No need to boot the playground if seamless.
-			return <JustViewport iframeRef={ref} />;
-		}
-		return (
-			<BrowserChrome
-				showAddressBar={!!playground}
-				url={url}
-				toolbarButtons={toolbarButtons}
-				onUrlChange={(url) => playground?.goTo(url)}
-				hideToolbar={hideToolbar}
-				className={className}
-			>
-				<JustViewport iframeRef={ref} />
-			</BrowserChrome>
-		);
+	if (displayMode === 'seamless') {
+		// No need to boot the playground if seamless.
+		return <JustViewport />;
 	}
-);
+	return (
+		<BrowserChrome
+			showAddressBar={!!playground}
+			url={url}
+			toolbarButtons={toolbarButtons}
+			onUrlChange={(url) => playground?.goTo(url)}
+			hideToolbar={hideToolbar}
+			className={className}
+		>
+			<JustViewport />
+		</BrowserChrome>
+	);
+};
 
-interface JustViewportProps {
-	iframeRef: Ref<HTMLIFrameElement>;
-}
+export const JustViewport = function LoadedViewportComponent() {
+	const iframeRef = useRef<HTMLIFrameElement>(null);
+	const iframe = (iframeRef as any)?.current;
+	useEffect(() => {
+		if (iframe) {
+			setupPostMessageRelay(iframe, document.location.origin);
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [!!iframe]);
```

Comment:
Does converting to boolean create a bug?

AFAICT, if the iframe instance ever changes, setup needs to happen again because listeners added by setupPostMessageRelay() contain conditions based on the specific instance like:
```js
if (event.source !== nestedFrame.contentWindow) {
	return;
}
```

Reply:
It did create a bug and I adjusted it. Great spot, thank you!